metabase/metabase
# Implement PoC and receive feedback

PoC for https://github.com/metabase/metabase/issues/28181

### Description

Preserve recent dashboard filter values

1. If there is no default value and user used some value on the dashboard, we need to restore this value when user gets back to the dashboard, no matter how user opens a dashboard (with a direct URL, after a search)
2. values are preserved per user per dashboard, we decided to use localStorage for it and if the idea works, we can involve BE to store those values on BE.
3. User needs to use the same machine and browser session to get benefits from this feature, preserving does not work if user switches machine and/or browser
4. Tests are red by intention. After the initial feedback from product and design they will be fixed

### How to verify

- Go to any dashboard with filters
- set value on the filter
- either remove URL search parameters (after `?`) or go to the root collection
- then find your dashboard and with filters and make sure that 
a. values are preserved
b. query for the mapped questions includes preserved parameter value into account

### Demo

https://www.loom.com/share/b6a895c2e722439eb75c67e49e845cae

### Checklist
